public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -19,8 +19,20 @@ import { ApiKeysActionBar } from './ApiKeysActionBar';
 import { ApiKeysTable } from './ApiKeysTable';
 import { ApiKeysController } from './ApiKeysController';
 import { ShowModalReactEvent } from 'app/types/events';
+import { contextSrv } from 'app/core/core';
 
 function mapStateToProps(state: StoreState) {
+  let canRead = true;
+  let canDelete = true;
+  let canCreate = true;
+  if (contextSrv.licensedAccessControlEnabled()) {
+    canRead = contextSrv.hasPermission(AccessControlAction.ActionAPIKeysRead);
+    canCreate = contextSrv.hasPermission(AccessControlAction.ActionAPIKeysCreate);
+    canDelete = contextSrv.hasPermission(AccessControlAction.ActionAPIKeysDelete);
+  }
```

Comment:
nit: you could use `contextSrv.hasAccess` to do this a bit more concisely (it'd be one call, and you'd use `true` as the fallback).

Reply:
One difference is that it seems `hasAccess` checks only for the feature flag, and not for the valid license. afaik we agreed that permission evaluation should work if you don't have valid license, so I assume it should be OK. So I am going to change as suggested, let me know if you think we do need license check also.

Reply:
Ah, good point. I think it should still work fine with an expired license in this case, but it's a good distinction to keep in mind.